### DISCUSSION TITLE
Blink: hit objects only update every measure and show for a short time

### DISCUSSION THREAD

**Discussion by Doksha** [ppy/osu#19788] [Ideas]
Every measure the hit objects update and then fade out (like hidden). hit objects would stay on screen for a proportion of the time (lets say 20%) to the next measure. This mod would add a new reading challenge, and would be available in all gamemodes. would also be very good for performance as only one frame would have to be rendered per measure (for the hit objects at least, cursor would still update every frame)

Customization options would be 
blink rate (every other measure, every measure, every half measure, and every quarter measure)
and
fade time (% of time to the next measure until the hit objects fade, anywhere from 0-100)

Mod would be accompanied by an epilepsy warning every time you activate it for obvious reasons.

Should probably be incompatible with HD (and flashlight for non standard gamemodes) Although it would be its own challenge when combined with either, and tbh not that bad

**Comment by mk56-spn** [comment]
this sounds rather enticing, and could have interesting dynamics when combined with flashlight for example. with people pre aiming a hard area to make it easier once the circles flash on screen

**Comment by mk56-spn** [comment]
https://user-images.githubusercontent.com/74463310/188998194-4fe2d4e8-732d-4cb1-9e07-a78a1e2e3c71.mov


https://user-images.githubusercontent.com/74463310/188997207-3ec55a3c-06fe-4f83-ad1f-04e0cffae466.mov

i have a WIP version of this coded. needs more work but leaving the clip here incase anyone wants to see it

**Reply by iSimpFor** [reply]
maybe add a mod setting to display the next measure of circles except darker? to have a preview of the next set of notes to hit

**Comment by mk56-spn** [comment]
Alright i think its done now.

went over my internal list of issues with the mod. ill wait for a thumbs up or some acknowledgement from the core Devs that this is indeed wanted before PRing it ( i saw smoogi upvoted the previous post i made on this but just double checking ) 

_i haven't added the suggestion by  iSImpFor in the comments above in the end to avoid bloat, if its wanted later down the line it can be done_

heres a very long video to show it works well with BPM changes on a marathon map( hidden is used since it helps show the effect of the mod) 

https://user-images.githubusercontent.com/74463310/190626839-989b8948-7dc1-4ab4-91d7-5fa885f841a7.mov

**Reply by smoogipoo** [reply]
Looks good!
